shell/utils/inactivity.ts

```diff
@@ -0,0 +1,183 @@
+};
+
+let storedUserActivityResponse: userActivityResponse = {
+  metadata: { name: '' },
+  status:   { expiresAt: '' }
+};
+
+let sessionTokenName: string;
+
+export async function checkBackendBasedSessionIdle(store: any): Promise<parsedInactivitySetting> {
+  let userActivity;
+  const canListSettings = store.getters[`management/canList`](MANAGEMENT.SETTING);
+  const canListUserAct = store.getters[`management/canList`](EXT.USERACTIVITY);
+  const canListTokens = store.getters[`rancher/canList`](NORMAN.TOKEN);
+
+  if (canListUserAct && canListTokens && canListSettings) {
+    const hash = {
+      userSessionTtlIdleSetting: store.dispatch('management/find', { type: MANAGEMENT.SETTING, id: SETTING.AUTH_USER_SESSION_IDLE_TTL_MINUTES }),
+      userSessionTtlSetting:     store.dispatch('management/find', { type: MANAGEMENT.SETTING, id: SETTING.AUTH_USER_SESSION_TTL_MINUTES }),
+      tokens:                    store.dispatch('rancher/findAll', { type: NORMAN.TOKEN }),
+    };
+
+    const res = await allHash(hash);
+
+    const userSessionTtlIdleSetting = res.userSessionTtlIdleSetting;
+    const userSessionTtlSetting = res.userSessionTtlSetting;
+    const tokens = res.tokens;
+
+    const ttlIdleValue = parseInt(userSessionTtlIdleSetting?.value || 0);
+    const ttlValue = parseInt(userSessionTtlSetting?.value || 0);
+
+    const sessionToken = tokens.find((token: any) => {
+      // this can be improved once https://github.com/rancher/rancher/issues/51580 is fixed
+      // we should not need to store the UI session token name globally
+      if (sessionTokenName) {
+        return token.name === sessionTokenName;
+      } else {
+        return token.description === 'UI session' && token.current;
+      }
+    });
+
+    // we only consider this feature as enabled IF ttlIdleValue < ttlValue (will be saving XHR requests if they are the same value - "normal" TTL will log out user)
+    if (sessionToken && ttlIdleValue < ttlValue) {
+      sessionTokenName = sessionToken.name;
+
+      try {
+        let requiresFreshData = false;
+
+        if (storedUserActivityResponse.status?.expiresAt) {
+          const currDate = Date.now();
+          const endDate = new Date(storedUserActivityResponse?.status?.expiresAt).getTime();
+
+          if (currDate < endDate) {
+            userActivity = storedUserActivityResponse;
+          } else {
+            requiresFreshData = true;
+          }
+        } else {
+          requiresFreshData = true;
+        }
+
+        // this is the POST to the UserActivity, which will reset the expireAt
+        if (requiresFreshData) {
+          userActivity = await updateUserActivityToken(store, sessionTokenName);
+        }
+
+        if (userActivity) {
+          const data = parseTTLData(userActivity);
+
+          console.debug(`UI inactivity modal (backend-based) will show after ${ data.showModalAfter || 0 / 60 }(m) and be shown for ${ data.courtesyTimer || 0 / 60 }(m)`); // eslint-disable-line no-console
```

Comment:
I just wanted to double check that you wanted to leave this in.

Also, I'm not sure how 0 / 60 is helping here.

Reply:
math was wrong because it was missing a pair of brackets... should be fixed